torch/onnx/_internal/diagnostics/rules.yaml

```diff
@@ -313,6 +313,32 @@
     deprecated: false
     tags: []
 
+- id: FXE0013
+  name: fx-node-explicit-type-promotion
+  short_description:
+    text: Determine if type promotion is required for the FX node. Explicitly insert cast nodes if needed.
+  full_description:
+    text: "Determine if type promotion is required for the FX node. Explicitly insert cast nodes if needed."
+    markdown: |
+      This diagnostic tracks the process of node level explicit type promotion. Implicit [type promotion](https://pytorch.org/docs/stable/tensor_attributes.html#type-promotion-doc) is a
```

Comment:
I think we should have a better explanation here on "it's needed to be explicit in ONNX, because it's implicit in PyTorch.". It seems to me the explanation here is a bit confusing.

Reply:
The same idea goes to the name itself. When we say: fx-node-explicit-type-promotion, it means we need explicit type promotion for ONNX. But it could be just me...

Reply:
Is `fx-node-insert-explicit-type-promotion` better? Or just `fx-node-insert-type-promotion`

Reply:
I'm leaving `explicit` in most of the places and all the names.